metadata-ingestion/src/datahub/ingestion/source/datahub/datahub_source.py

```diff
@@ -48,54 +48,70 @@ def get_workunit_processors(self) -> List[Optional[MetadataWorkUnitProcessor]]:
-        stop_time = datetime.now(tz=timezone.utc)
-        mcls = DataHubKafkaReader(self.config, self.report).get_mcls(
-            from_offset=from_offset, stop_time=stop_time
-        )
-        for i, (mcl, offset) in enumerate(mcls):
-            # TODO: Get rid of deserialization?
-            mcp = MetadataChangeProposalWrapper.try_from_mcl(mcl)
-            if isinstance(mcp, MetadataChangeProposalWrapper):
-                yield mcp.as_workunit()
-            else:
-                yield MetadataWorkUnit(
-                    id=f"{mcp.entityUrn}-{mcp.aspectName}-{i}", mcp_raw=mcp
+            self.report.num_mysql_aspects_ingested += 1
+
+            if (
+                self.config.commit_with_parse_errors
+                or not self.report.num_mysql_parse_errors
+            ):
+                self.stateful_ingestion_handler.update_checkpoint(
+                    last_createdon=createdon
                 )
-
-            self.stateful_ingestion_handler.update_checkpoint(last_offset=offset)
             self._commit_progress(i)
 
+    def _get_kafka_workunits(
+        self, from_offsets: Dict[int, int], stop_time: datetime
+    ) -> Iterable[MetadataWorkUnit]:
+        logger.info(f"Fetching timeseries aspects from kafka until {stop_time}")
+
+        with DataHubKafkaReader(self.config, self.report, self.ctx) as reader:
+            mcls = reader.get_mcls(from_offsets=from_offsets, stop_time=stop_time)
+            for i, (mcl, offset) in enumerate(mcls):
+                mcp = MetadataChangeProposalWrapper.try_from_mcl(mcl)
+                if isinstance(mcp, MetadataChangeProposalWrapper):
+                    yield mcp.as_workunit()
+                else:
+                    yield MetadataWorkUnit(
+                        id=f"{mcp.entityUrn}-{mcp.aspectName}-{i}", mcp_raw=mcp
+                    )
+                self.report.num_kafka_aspects_ingested += 1
+
+                if (
+                    self.config.commit_with_parse_errors
+                    or not self.report.num_kafka_parse_errors
+                ):
+                    self.stateful_ingestion_handler.update_checkpoint(
+                        last_offset=offset
+                    )
+                self._commit_progress(i)
+
     def _commit_progress(self, i: Optional[int] = None) -> None:
         """Commit progress to stateful storage, if there have been no errors.
 
         If an index `i` is provided, only commit if we are at the appropriate interval
         as per `config.commit_state_interval`.
         """
-        has_errors = (
-            self.report.num_mysql_parse_errors or self.report.num_kafka_parse_errors
-        )
         on_interval = (
             i
             and self.config.commit_state_interval
             and i % self.config.commit_state_interval == 0
         )
 
-        if not has_errors and (i is None or on_interval):
+        if i is None or on_interval:
```

Comment:
so commits happen regardless of `commit_with_parse_errors`, while updating the state only happens conditionally?

Reply:
Yeah, I swapped it cause this logic seemed simpler -- this allows me to update the kafka state if there are only mysql errors, and vice versa.

Reply:
makes sense